### ISSUE TITLE
Please provide some substance to the claims made in the README

### ISSUE THREAD

**Issue by Licenser** [redox-os/tfs#66]
The promise made in the README is exciting however at the end it leaves a bit of an empty feeling.

In the end, for a technical audience, it shoulds a lot like marketing and has the bad aftertaste of the btrfs disaster.

Between all the claims made of capabilities of TFS none seems to be backed up by research, or studies, or benchmarks, or really anything.

Comparisons with other filesystems seem to lack research on the topic and instead of actually comparing them are stated as broad groupings `"some other"`, `"few filesystems"`.

Estimates are presented without any base on what they are based on, not to mention example cases that would back them up. (i.e. `"It is estimated that you get 60-120% more usable space."`)

Parts are worded misleadingly and could easily be misinterpreted to mean more than they actually do (i.e. `"TFS stores a revision history of every file without imposing extra overhead"` which I suspect means overhead beyond the delta as there needs to be overhead compared to the un-revisioned file).

`All memory safe` is outright false and self-contradicting in the explanation. 

All this seems to hurt the credibility of the implementation especially when the claim is that `"many components are complete"` at which point it seems sensible to expect some backing of the claims based on actual data.

**Comment by Licenser** [comment]
The `Full coverage testing` section also is somewhat questionable, there seem to be no automated tests (at least not publically visible). Manually running fails a few or parts of the codebase have none at all (`mlcr`). In addition, there is research to suggest that there test coverage has a significant (let alone strong) impact on the quality of tests (http://www.linozemtseva.com/research/2014/icse/coverage/coverage_paper.pdf)

**Comment by ticki** [comment]
It is all under "design **goals**", but it is somewhat unclear, so yeah, it's misleading. I plan to rewrite the README entirely eventually.

**Comment by Licenser** [comment]
I think what makes it misleading is that it's worded as facts i.e. `"TFS is the first file system to incorporate complete full-disk..."`, `"TFS stores a revision history of every file without imposing extra overhead..."` and so on rather than stating goals i.e. `"TFS wants to incorporate complete full-disk..."`, `"TFS aims to store a revision history of every file without imposing extra overhead"`

But even when reading the sectio as pure goals instead of facts, a bit on the 'how' would go a long way for credibility.

**Comment by ticki** [comment]
Fair point.

**Comment by brokenpeace** [comment]
Perhaps instead of writing about the lacking information and benchmarks (which take time to critique) spend that time helping the project?
Open source is a contribution of donated time from us all. Dislike something? Missing something? Do it! Submit your PR let's get moving?

**Comment by leeoniya** [comment]
while @ticki certainly does not owe anyone anything, extraordinary claims *do* require extraordinary evidence. being an open source project done in free time does nothing to lift this burden. rewording the claims is certainly far easier than providing exhaustive and good faith proof that account for all contributing factors. few people besides the author can meaningfully contribute such things.

**Comment by ticki** [comment]
I agree with @leeoniya. A good project should provide sources for claims, and it is pretty unclear that these are goals at the moment, especially in how I've worded things.

**Comment by Licenser** [comment]
@brokenpeace Code is not the only kind of contribution to open source, as you said yourself I did spend time on writing this issue, analyzing where things are off and presenting it in a constructive way.
